src/test/java/com/breskul/bibernate/persistence/EntityManagerImplTest.java

```diff
@@ -383,4 +387,48 @@ public void testFindMethodWithManyToOneRelationRelatedObjectIsNull() {
         entityManager.getTransaction().commit();
     }
 
+    @Test
+    @DisplayName("Test lazy loading. Get list of related objects out of transaction")
+    public void testLazyLoading() {
+
+        PersonWithoutGeneratedValue person = new PersonWithoutGeneratedValue();
+        person.setFirstName("Quentin");
+        person.setLastName("Tarantino");
+        person.setId(40L);
+        person.setBirthday(LocalDateTime.of(1963, Month.MARCH, 27, 10, 0, 0).toLocalDate());
+
+        NoteWithoutGeneratedValue firstNote = new NoteWithoutGeneratedValue();
+        firstNote.setId(51L);
+        firstNote.setBody("Test note #1");
+        firstNote.setPerson(person);
+
+        NoteWithoutGeneratedValue secondNote = new NoteWithoutGeneratedValue();
+        secondNote.setId(52L);
+        secondNote.setBody("Test note #2");
+        secondNote.setPerson(person);
+
+        person.addNote(firstNote);
+        person.addNote(secondNote);
+
+        entityManager.getTransaction().begin();
+        entityManager.persist(person);
+        entityManager.getTransaction().commit();
+
+        EntityManagerImpl otherEntityManager = new EntityManagerImpl(dataSource);
+        otherEntityManager.getTransaction().begin();
+        var selectedPerson = otherEntityManager.find(PersonWithoutGeneratedValue.class, person.getId());
+        otherEntityManager.getTransaction().commit();
```

Comment:
if we add here Line of code `selectedPerson.getNotes().forEach(System.out::println);` then test will never pass. this happens because method `DaoUtils.getColumnName` does not account for `JoinColumn` . Hence, internally we will receive sql exception from lazy list because query `select * from notes where person = ?` is not valid. I have submitted commit for` DaoUtils.getColumnName`. When adding  

```
//        if (field.isAnnotationPresent(JoinColumn.class)){
//            return field.getAnnotation(JoinColumn.class).name();
//        }
```
lazy list works fine

Reply:
agree